e2e/payments.py

```
@@ -0,0 +1,18 @@
+import hashlib
+
+
+def lookup_user(conn, user_id):
+    # SQL injection: user_id is interpolated straight into the statement.
+    cur = conn.cursor()
+    cur.execute("SELECT * FROM users WHERE id = '" + user_id + "'")
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)